src/librustc_middle/query/mod.rs

```diff
@@ -214,6 +214,12 @@ rustc_queries! {
             cache_on_disk_if { key.is_local() }
         }
 
+        query coverage_data(key: DefId) -> Option<mir::CoverageData> {
```

Comment:
When you make this query operate by walking the `Body` MIR, I think it would be reasonable to change the return type to just `mir::CoverageData`. The query should be able to operate on any `Body` even ones that aren't instrumented since it will just return zero `num_counters`.

Reply:
Makes sense to me. Note, `num_counters == 0` implies there are no calls to `instrprof.increment`, and therefore no need for the `hash` argument for the given function, so I will also set the hash to `0` in that case.

Reply:
Done. PTAL. Thanks!